src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -0,0 +1,440 @@
+          );
+          ownerDocument.removeEventListener(
+            handlersAndEvents.touch.upEvent,
+            handlersAndEvents.touch.moveHandler,
+          );
+          dispatchEnd();
+        },
+      },
+    };
+
+    const events = isTouchEvent
+      ? handlersAndEvents.touch
+      : handlersAndEvents.mouse;
+    const passiveIfSupported = passiveEventSupported()
+      ? { passive: false }
+      : false;
+    ownerDocument.addEventListener(
+      events.moveEvent,
+      events.moveHandler,
+      passiveIfSupported,
+    );
+    ownerDocument.addEventListener(
+      events.upEvent,
+      events.upHandler,
+      passiveIfSupported,
+    );
+
+    dispatch({
+      type: actions.columnStartResizing,
+      columnId: header.id,
+      columnWidth: header.width,
+      nextColumnWidth: nextHeader?.width,
+      headerIdWidths,
+      nextHeaderIdWidths,
+      clientX,
+    });
+  };
+
+  return [
+    props,
+    {
+      onMouseDown: (e: React.TouchEvent & React.MouseEvent) => {
+        e.persist();
+        e.stopPropagation();
+        onResizeStart(e, header);
+      },
+      onTouchStart: (e: React.TouchEvent & React.MouseEvent) => {
+        e.persist();
+        e.stopPropagation();
+        onResizeStart(e, header);
+      },
+      style: {
+        cursor: 'col-resize',
+      },
+      draggable: false,
+      role: 'separator',
+    },
+  ];
+};
+
+useResizeColumns.pluginName = 'useResizeColumns';
+
+const reducer = <T extends Record<string, unknown>>(
+  newState: TableState<T>,
+  action: ActionType,
+  previousState: TableState<T>,
+  instance?: TableInstance<T>,
+) => {
+  if (action.type === actions.init) {
+    return {
+      ...newState,
+      columnResizing: {
+        columnWidths: {},
+      },
+    };
+  }
+
+  if (action.type === actions.resetResize) {
+    return {
+      ...newState,
+      columnResizing: {
+        columnWidths: {},
+      },
+    };
+  }
+
+  if (action.type === actions.columnStartResizing) {
+    const {
+      clientX,
+      columnId,
+      columnWidth,
+      nextColumnWidth,
+      headerIdWidths,
+      nextHeaderIdWidths,
+    } = action;
+
+    return {
+      ...newState,
+      columnResizing: {
+        ...newState.columnResizing,
+        startX: clientX,
+        columnWidth,
+        nextColumnWidth,
+        headerIdWidths,
+        nextHeaderIdWidths,
+        isResizingColumn: columnId,
+      },
+    };
+  }
+
+  if (action.type === actions.columnResizing) {
+    const { clientX } = action;
+    const {
+      startX = 0,
+      columnWidth = 1,
+      nextColumnWidth = 1,
+      headerIdWidths = [],
+      nextHeaderIdWidths = [],
+    } = newState.columnResizing;
+
+    const deltaX = clientX - startX;
+
+    const newColumnWidths = getColumnWidths(
+      headerIdWidths,
+      deltaX / columnWidth,
+    );
+    const newNextColumnWidths = getColumnWidths(
+      nextHeaderIdWidths,
+      -deltaX / nextColumnWidth,
```

Comment:
From what I understand, this approach is to preserve the total width of the table.

What if the table is unbounded and needs to be resized beyond the screen?

Reply:
Devs can increase table width or ask us for resize mode 😀